### PR TITLE
[NavigationBar] - Added Accessibility Mutator

### PR DESCRIPTION
**Pull request by ShepJGoogle**
Added initial implementation of MDCNavigationBarAccessibilityEnforcer

- Created a enforcer class for MDCNavigationBar
- Implemented config method to update colors on MDCNavigationBar's title and bar button items

Design Doc: https://docs.google.com/a/google.com/document/d/1QHdBcW8pIkmJ796ou_W-wIznLXT_vzcxWwwlqSSN9x8/edit?usp=sharing

### CODE REVIEW

**Comment by willlarche** [comment]
Is this blocked?

**Comment by ShepJGoogle** [comment]
No I think just waiting on review of the design doc attached and the changes.

File: components/NavigationBar/src/MDCNavigationBar.h [hunk 1]
```diff
@@ -44,6 +44,31 @@ typedef NS_ENUM(NSInteger, MDCNavigationBarTitleAlignment) {
...
+
+@interface MDCNavigationBarAccessibilityEnforcer : NSObject
+
+- (nonnull instancetype)init;
+
+/**
+  Mutates title text color and navigation items' tint colors based on background color of
+  navigation bar.
+  */
+- (void)enforceFontColorAccessibility:(nonnull MDCNavigationBar *)navBar;
```
Comment (randallli) [inline components/NavigationBar/src/MDCNavigationBar.h]: enforceTextColor

**Comment by ianegordon** [comment]
Does this work with the UIAppearance proxy?  (Is that a concern?)

File: components/NavigationBar/src/MDCNavigationBar.h [hunk 2]
```diff
@@ -44,6 +44,31 @@ typedef NS_ENUM(NSInteger, MDCNavigationBarTitleAlignment) {
...
+  an instance of a Material Navigation Bar to activate and esnure accessibility on its title and 
+  buttons.
+
+  ### Dependencies
+
+  Material NavigationBarAccessibilityEnforcer depends on the Navigation Bar material component and
+  MDFTextAccessibility Framework.
+  */
+
+@interface MDCNavigationBarAccessibilityEnforcer : NSObject
```
Comment (randallli) [inline components/NavigationBar/src/MDCNavigationBar.h]: MDCNavigationBarTextColorAccessibilityMutator

File: components/NavigationBar/src/MDCNavigationBar.h (hunk 1 above)
Comment (randallli) [reply components/NavigationBar/src/MDCNavigationBar.h]: actually can we just change this to:
- (void)mutate:(nonnull MDCNavigationBar *)navBar;
or
- (void)mutate:(id)navBar;

this way other mutators can have the same api signature.

**Review by randallli — CHANGES_REQUESTED** [review]

File: components/NavigationBar/examples/NavigationBarTypicalUseExample.m [hunk 3]
```diff
@@ -34,11 +34,12 @@ - (void)viewDidLoad {
 
   self.navBar = [[MDCNavigationBar alloc] initWithFrame:CGRectZero];
   [self.navBar observeNavigationItem:self.navigationItem];
-  self.navBar.tintColor = [UIColor whiteColor];
-  self.navBar.titleTextAttributes = @{NSForegroundColorAttributeName : [UIColor whiteColor]};
 
   // Light blue 500
   [self.navBar setBackgroundColor:[UIColor colorWithRed:0.012 green:0.663 blue:0.957 alpha:1]];
+  MDCNavigationBarAccessibilityEnforcer *enforcer =
```
Comment (randallli) [inline components/NavigationBar/examples/NavigationBarTypicalUseExample.m]: these need to be changed too i think

**Review by randallli — APPROVED** [review]

**Comment by ShepJGoogle** [comment]
@ianegordon @randallli and I just did some investigation and UIAppearance proxy returns an instancetype of whatever class you call it on. As such, passing this instance as an argument of the mutate: does in fact mutate the proxy.

**Comment by randallli** [comment]
merge when your ready.

**Pull request merged**